gensim/models/nmf.py

```diff
@@ -318,11 +316,13 @@ def update(self, corpus, chunks_as_numpy=False):
                 if self._R is not None:
                     self._R.append(r)
 
+                self.A *= chunk_idx - 1
```

Comment:
Are you familiar with scipy's `sparsetools`?

We're using it to good effect in Gensim already: [example link](https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/models/lsimodel.py#L926).

It's more low level and avoids some unnecessary memory copies, and allows operations that are harder to do through the top-level "matrix API".

Reply:
Thanks, I'll look into it!